nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/CoreSession.java

```diff
@@ -167,6 +167,13 @@
     boolean hasPermission(Principal principal, DocumentRef docRef, String permission);
 
     /**
+     * Filters the supplied permissions base on whether they are granted to a given principal for a given document.
+     *
+     * @since 9.1
+     */
+    String[] filterGrantedPermissions(Principal principal, DocumentRef docRef, String... permissions);
```

Comment:
For new APIs please try to use `List<String>` instead of old-style arrays. And take a `List<String>` instead of varargs, unless you have a good justification for varargs :)